### PR TITLE
Fix/upload artifact error windows

### PR DESCRIPTION
**Pull request by fuxiaohei**
From issue https://github.com/go-gitea/gitea/issues/27314

When act_runner in `host` mode on Windows. `upload_artifact@v3` actions use `path.join` to generate `itemPath` params when uploading artifact chunk. `itemPath` is encoded as `${artifact_name}\${artifact_path}`.  

<del>It's twice query escaped from ${artifact_name}/${artifact_path} that joined by Windows slash \.</del>

**So we need convert Windows slash to linux**.

In https://github.com/go-gitea/gitea/issues/27314, runner shows logs from  `upload_artifact@v3` like with `%255C`:

```
[artifact-cases/test-artifact-cases]   | ::error::Unexpected response. Unable to upload chunk to http://192.168.31.230:3000/api/actions_pipeline/_apis/pipelines/workflows/6/artifacts/34d628a422db9367c869d3fb36be81f5/upload?itemPath=more-files%255Css.json
```

But in gitea server at the same time, But shows `%5C`

```
2023/10/27 19:29:51 ...eb/routing/logger.go:102:func1() [I] router: completed PUT /api/actions_pipeline/_apis/pipelines/workflows/6/artifacts/34d628a422db9367c869d3fb36be81f5/upload?itemPath=more-files%5Css.json for 192.168.31.230:55340, 400 Bad Request in 17.6ms @ <autogenerated>:1(actions.artifactRoutes.uploadArtifact-fm)
```

I found `%255C` is escaped by `https://github.com/actions/upload-artifact/blob/main/dist/index.js#L2329`.

### CODE REVIEW

**Comment by KN4CK3R** [comment]
Isn't that something the runner should do?

**Comment by silverwind** [comment]
Double encoding would definitely be a bug in the runner.

Also `util.PathJoinRel` seems to be meant for joining 2 or more paths. calling it on a single path seems wrong and I think `path.Clean` would achieve the same in a cleaner manner.

**Comment by wxiaoguang** [comment]
It's not right to do `url.QueryUnescape(ctx.Req.URL.Query().Get("itemPath"))` (double decoding)

I think it should figure out the runner's bug first.

**Comment by fuxiaohei** [comment]
> It's not right to do `url.QueryUnescape(ctx.Req.URL.Query().Get("itemPath"))` (double decoding)
> 
> I think it should figure out the runner's bug first.

I make a mistake that confusing logs from `upload_artifact@v3`. I rewrite the description of this bug above.

File: routers/api/actions/artifacts_utils.go [hunk 1]
```diff
@@ -58,7 +57,8 @@ func validateArtifactHash(ctx *ArtifactContext, artifactName string) bool {
 func parseArtifactItemPath(ctx *ArtifactContext) (string, string, bool) {
 	// itemPath is generated from upload-artifact action
 	// it's formatted as {artifact_name}/{artfict_path_in_runner}
-	itemPath := util.PathJoinRel(ctx.Req.URL.Query().Get("itemPath"))
+	// act_runner in host mode on Windows, itemPath is joined by Windows slash '\' with %5C urlencoded format
+	itemPath := strings.ReplaceAll(ctx.Req.URL.Query().Get("itemPath"), "\\", "/")
```
Comment (wxiaoguang) [inline routers/api/actions/artifacts_utils.go]:
```suggestion
	// act_runner in host mode on Windows, itemPath is joined by Windows slash '\'
	itemPath := util.PathJoinRelX(ctx.Req.URL.Query().Get("itemPath"))
```

**Comment by wxiaoguang** [comment]
> Double encoding would definitely be a bug in the runner.
> 
> Also `util.PathJoinRel` seems to be meant for joining 2 or more paths. Calling it with a single path seems wrong and I think `path.Clean` would achieve the same in a cleaner manner.

```
// PathJoinRel joins the path elements into a single path, each element is cleaned by path.Clean separately.

// PathJoinRelX joins the path elements into a single path like PathJoinRel,
// and covert all backslashes to slashes. (X means "extended", also means the combination of `\` and `/`).
```


`PathJoinRelX` is the right function to call.

**Review by wxiaoguang — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Pull request merged**